source/rafcon/core/storage/storage.py

```diff
@@ -369,6 +373,48 @@ def load_state_machine_from_path(base_path, state_machine_id=None):
     return state_machine
 
 
+def reconnect_data_flow(state_machine):
+    queue = [state_machine.root_state]
+    while len(queue) > 0:
+        state = queue.pop(0)
+        if hasattr(state, 'is_dummy') and state.is_dummy:
+            same_level_states = [state.parent]
+            for same_level_state in state.parent.states.values():
+                if same_level_state.state_id != state.state_id:
+                    same_level_states.append(same_level_state)
+            for same_level_state in same_level_states:
+                for data_flow in state.parent.data_flows.values():
+                    data_type = int
```

Comment:
This `reconnect_data_flow` function does not handle scoped variables. This is probably also why `data_type` is here set to `int` by default, because in the following, it is only checked if the data flow is connected to an input or output port, not to a scoped variable. If also the last case was handled, it wouldn't be necessary to initialize `data_type` and `default_value` with values.

I also verified this again with the `wait` library. When connecting the input of this library to a scoped variable of the parental state and I rename the wait state, the data type of the input port of the dummy state is int instead of float.

Reply:
Nice catch! Fixed. Still, I initialize data_type & default_value just to be on the safe side.

Reply:
Thank you for quickly fixing this!

It is a good style in software development, to always have an `else` case. There you can catch errors, and e.g. print an error message, as this is a case that is not expected, so might point to an error at another location in the code.